web/stream/pool.go

```diff
@@ -0,0 +1,86 @@
+package stream
+
+import (
+	"errors"
+	"sync"
+)
+
+// Pool is a pool of channels.
+// The channels are used to send data to connected clients.
+type Pool[T any] struct {
```

Comment:
The `Pool` type should have a `pool_test.go` that tests the API, including adding and removing users, sending stuff on its channels and more.

Also, make sure that all methods in the API are used (didn't check yet). If we don't need, e.g., `CloseBy`, we shouldn't add it until we need it.

Reply:
Do we need to export this type and the methods? Looks like they are only being used in the `stream` package.

Reply:
Also, `CloseBy` does not make sense to keep after giving it a second thought. I cannot see any cases where we would want to only close a channel (signalling to a stream that it should close), but still keep it in our map of channels.